### PR TITLE
Refactor: Device refactor to integrate Platform-Devices

### PR DESCRIPTION
**Pull request by IgnoreWarnings**
Rename (pci::)Device to PciDevice and move it to devices folder. Change DeviceList to PciDeviceList. Change namespace from pci to devices.
#808 Introduces a Device interface which will integrate Pci Devices and Platform Devices.

### CODE REVIEW

**Review by n-eiling — APPROVED** [review]

**Comment by n-eiling** [comment]
@stv0g we are waiting for your approval :)

**Comment by stv0g** [comment]
Glaube da sind irgendwie Zertifikate in der CI abgelaufen:

```
tls: failed to verify certificate: x509: certificate has expired or is not yet valid: current time 2024-08-29T15:02:11Z is after 2024-08-29T08:34:57Z
```

File: common/lib/CMakeLists.txt
Comment (stv0g) [inline common/lib/CMakeLists.txt:42]: Would it be okay, if we rename it to `kernel/devices/pci.h`? I dislike the duplication of `device` in the path.

File: fpga/lib/pcie_card.cpp
Comment (stv0g) [inline fpga/lib/pcie_card.cpp:28]: Isnt there one `(` too much?

**Review by stv0g — CHANGES_REQUESTED** [review]
Looks good to me :) Could you rename `pci_device.hpp` to just `pci.hpp`, fix the CI and remove to one superfluous parantheses?

Thanks :) 👍🏻

**Comment by n-eiling** [comment]
I guess it was fixed. I fixed the CI by rerunning it.

File: common/lib/CMakeLists.txt
Comment (stv0g) [reply common/lib/CMakeLists.txt:42]: Okay, good argument. Lets keep `pci_device.hpp` then

File: fpga/lib/pcie_card.cpp
Comment (stv0g) [reply fpga/lib/pcie_card.cpp:28]: Could you fix it?

**Review by stv0g — APPROVED** [review]

File: fpga/lib/pcie_card.cpp
Comment (stv0g) [reply fpga/lib/pcie_card.cpp:28]: Nevermind, I am gonna merge this now. It would be nice if it could be addresses in a subsequent MR

**Pull request merged**
